### PR TITLE
Add aql and aqlQuery functions to type defs

### PR DESCRIPTION
**Pull request by jeckhart**
Add aql and aqlQuery functions to type defs and associated Database.query functions to type defs

### CODE REVIEW

**Comment by pluma** [comment]
Thank you for the PR. For legal reasons we need a signed Contributor License Agreement (CLA) before we can merge your code into the project.

You can download the form from our website: https://www.arangodb.com/documents/cla.pdf

Once you've filled out and signed it, please send it to us via e-mail: cla@arangodb.com or fax: +49-221-2722999-88.

File: arangojs.d.ts [hunk 1]
```diff
@@ -100,11 +118,14 @@ declare module "arangojs" {
          */
         query(query: string, bindVars?: any, opts?: any): Promise<Cursor>;
         /**
-         * Template string handler for AQL queries. Converts an ES2015 template string to an object that can be passed to `database.query` by converting arguments to bind variables
+         * Performs a database query using the interpolated query from the tagged template from aql or aqlQuery
+         * then returns a new `Cursor` instance for the result list
          *
-         * Any Collection instances will automatically be converted to collection bind variables
+         * @param query An AQL query string or a query builder instance
+         * @param bindVars An object defining the variables to bind the query to
```
Comment (pluma) [inline arangojs.d.ts]: The `@param` comments are a verbatim copy of the other signature. You might want to adjust that.

Comment (jeckhart) [inline arangojs.d.ts]: Thank you, updated

File: arangojs.d.ts [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+     *
+     * Any Collection instances will automatically be converted to collection bind variables
+     */
+    export function aqlQuery(strings: ReadonlyArray<string>, ...args): AQLQuery;
+    /**
+     * Template string handler for AQL queries. Converts an ES2015 template string to an object that can be passed to `database.query` by converting arguments to bind variables
+     *
+     * Any Collection instances will automatically be converted to collection bind variables
+     */
+    export function aql(strings: ReadonlyArray<string>, ...args): AQLQuery;
```
Comment (Fahrradflucht) [inline arangojs.d.ts]: Don't use untyped rest spread. This will fail if projects enable `noImplicitAny`.

File: arangojs.d.ts [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+      query: string;
+      bindVars: any;
+    }
+
+    /**
+     * Template string handler for AQL queries. Converts an ES2015 template string to an object that can be passed to `database.query` by converting arguments to bind variables
+     *
+     * Any Collection instances will automatically be converted to collection bind variables
+     */
+    export function aqlQuery(strings: ReadonlyArray<string>, ...args): AQLQuery;
```
Comment (Fahrradflucht) [inline arangojs.d.ts]: Don't use untyped rest spread. This will fail if projects enable `noImplicitAny`.

**Comment by pluma** [comment]
@jeckhart I still haven't received word that we have a CLA from you on file. I'm closing this in favour of #428 which seems related.

Thank you for the PR nevertheless.

**Comment by Fahrradflucht** [comment]
@pluma I don't think that #428 fixes this. I only commented here to ensure that this doesn't cause a regression if because it introduces even more untyped reset spreads.

**Comment by pluma** [comment]
I think you're right. Sadly I can't merge this without a CLA either way :neutral_face:

**Pull request closed without merging**